test/tools/scripts/extract-testcases.py

```diff
@@ -0,0 +1,227 @@
+                        evaluated = eval(arg)
+                        if type(evaluated) is int:
+                            if string.find("0x") >= 0:
+                                r = hex(evaluated)
+                            else:
+                                r = int(evaluated)
+                        elif type(evaluated) is str:
+                            r = arg
+                        result += str(r) + ", "
+                    except:
+                        self.extractable = False
+                        result += "?, "
+            return result[:-2]
+        return ""
+
+    def create_call(self, signature, parameters, expectations, comment):
+        result = signature
+        if parameters:
+            result += ": "
+            result += parameters
+        result += " -> "
+        if expectations:
+            result += expectations
+        if comment:
+            result += " # " + comment + " #"
+        return result
+
+    def create_isoltest_call(self, line, test_comment):
+        search = re.search(r'ABI_CHECK\(callContractFunction\(\"(.*)\",((.|\s)*)\), encodeArgs\(((.|\s)*)\)\);', line,
+                           re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), self.eval_and_correct(search.group(2)),
+                                               self.eval_and_correct(search.group(4)), test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\(\"(.*)\",((.|\s)*)\), fromHex\("((.|\s)*)"\)\);', line,
+                           re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), self.eval_and_correct(search.group(2)),
+                                               self.eval_and_correct("0x" + search.group(4)), test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\("(.*)"\), encodeArgs\(((.|\s)*)\)\);', line, re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), "", self.eval_and_correct(search.group(2)),
+                                               test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\("(.*)"\), encodeDyn\(((.|\s)*)\)\);', line, re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), "",
+                                               self.eval_and_correct(search.group(2)), test_comment)
+            return result
+
+        search = re.search(r'ABI_CHECK\(callContractFunction\("(.*)"\), fromHex\("((.|\s)*)"\)\);', line, re.M | re.I)
+        if search:
+            result = self.create_call(search.group(1).strip(), "",
+                                               self.eval_and_correct(search.group(2)), test_comment)
+            return result
+
+        self.extractable = False
+        return "?"
+
+    def extract(self):
+        test_file = open(
```

Comment:
Can you check that it fails if the file already exists?

Reply:
I think it will just overwrite, but I'll check that